holoviews/core/util/dependencies.py

```diff
@@ -0,0 +1,118 @@
+import re
+import sys
+from functools import lru_cache
+from importlib import import_module
+from importlib.metadata import PackageNotFoundError, version
+from importlib.util import find_spec
+
+_re_no = re.compile(r"\d+")
+
+class VersionError(Exception):
+    """Raised when there is a library version mismatch."""
+
+    def __init__(self, msg, version=None, min_version=None, **kwargs):
+        self.version = version
+        self.min_version = min_version
+        super().__init__(msg, **kwargs)
+
+
+@lru_cache
+def _is_installed(module_name):
+    return find_spec(module_name) is not None
+
+
+@lru_cache
+def _get_version(package_name):
+    try:
+        return version(package_name)
+    except PackageNotFoundError:
+        return "0.0.0"
+
+
+def _no_import_version(package_name) -> tuple[int, int, int]:
+    """Get version number without importing the library"""
+    version_str = _get_version(package_name)
+    return tuple(map(int, _re_no.findall(version_str)[:3]))
+
+
+_MIN_SUPPORTED_VERSION = {
+    "pandas": (1, 3, 0),
+}
+
+
+class _lazy_module:
+    __slots__ = ("__bool_use_sys_modules", "__module", "__module_name", "__package_name")
```

Comment:
Seems unnecessary.

Reply:
It has been removed. 

![image](https://github.com/user-attachments/assets/c8afb995-b00e-411c-9091-ec9c0466e6bf)